### ISSUE TITLE
Drag and drop multiple documents

### ISSUE THREAD

**Issue by anthonyirwin82** [vkbo/novelWriter#1549]
When planing a story I create a bunch of Project Notes/Novel Documents for scene ideas, plot points, story beats or whatever term you want to use.

As the planing progresses, I would like to be able to select a number of documents at the same time and drag them into a folder.

This could be an act 1 folder earlier in the process and then chapter folders as I drill down in the planning.

At the moment, you need to move each document/note individually, which is time-consuming.

**Comment by vkbo** [comment]
Multi-document actions in the project tree is a work in progress. I've added the feature to drag or delete whole folders, but not yet to drag multiple individual documents.

In the mean time, I would suggest starting off with few documents and use the split feature to split them up as the story/outline grows. At least that's my workflow.

**Comment by vkbo** [comment]
So, I've been trying to crack multiple-document drag and drop for a bit now. It is a non-trivial problem, which is why I have avoided it so far. There are several issues to handle. One is that root folders cannot be dragged and dropped at all, but I've solved this by not allowing selecting root folders when multi-selecting. Any drag and drop event including root folders are also automatically cancelled.

But, the default drag and drop handling in the Qt framework has a concerning approach to drag and drop if you select both a folder and the files within it. The files are not moved along with the folder, but added below it. That's going to be annoying.

Also, the multi-select drag and drop means the undo functionality currently in the tree will no longer work either.
